plugins/libplugin.c

```diff
@@ -644,6 +645,11 @@ handle_getmanifest(struct command *getmanifest_cmd,
 		}
 		json_object_end(params);
 	}
+#if DEVELOPER
+	json_object_start(params, NULL);
+	json_add_string(params, "name", "memleak");
+	json_object_end(params),
+#endif /* DEVELOPER */
```

Comment:
This presumably forces a plugin to be recompiled if the `DEVELOPER` option doesn't match the one we compiled `lightningd` with. This seems suboptimal if we ever want to distribute pre-compiled plugins, for example as part of the reckless plugin manager.

Reply:
Would it make sense to always register this, but have stubs that do nothing if we aren't in `DEVELOPER=1` mode?

Reply:
Alternatively we could add a `shutdown` hook that is always present, allows plugins to flush their state instead of killing them outright, and piggyback some information back to `lightningd` (in this case the memleak info, getting ignored if `lightningd` wasn't compiled with `DEVELOPER=1`)

Reply:
Yeah, I like a shutdown hook better.  Let me do that instead!